safenode/src/node/api.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use std::time::Duration;
+
+use futures::future::select_all;
+use xor_name::XorName;
+
+use super::error::Result;
+use crate::{
+    network::{Cmd, Query, Request},
+    storage::chunks::Chunk,
+};
+
+use super::Node;
+
+impl Node {
+    pub async fn store_data(&self, data: Chunk) -> Result<()> {
+        info!("stroing data: {:?}", data.address());
+        self.storage.store(&data).await?;
+        let closest_nodes = self.network.get_closest_nodes(data.address().0).await?;
+        let mut sent_futures = Vec::new();
+        for node in closest_nodes {
+            let req = Request::Cmd(Cmd::StoreChunk(data.clone()));
+            let future = Box::pin(tokio::time::timeout(
+                Duration::from_secs(5),
+                self.network.send_request(req, node),
+            ));
+            sent_futures.push(future);
+        }
+
+        let mut responses = Vec::new();
+        while !sent_futures.is_empty() {
```

Comment:
we may don't care of the response of `store data` ?
anyway, no harm to keep it unless it blocks the flow or affect the performance